src/argowrapper/routes/routes.py

```diff
@@ -153,11 +182,13 @@ def submit_workflow(
     request_body: Dict[Any, Any],
     request: Request,  # pylint: disable=unused-argument
 ) -> str:
-    """route to submit workflow"""
+    """check if user has a billing id tag"""
+    billing_id = check_user_billing_id(request)
```

Comment:
please move this line inside `try` so that any exception can be converted to the HTTP500 error in the `except` part below.